ospf6d/ospf6_bfd.c

```diff
@@ -118,134 +87,37 @@ void ospf6_bfd_trigger_event(struct ospf6_neighbor *on, int old_state,
  *                               zebra for starting/stopping the monitoring of
  *                               the neighbor rechahability.
  */
-static void ospf6_bfd_reg_dereg_all_nbr(struct ospf6_interface *oi, int command)
+static void ospf6_bfd_reg_dereg_all_nbr(struct ospf6_interface *oi, bool enable)
 {
 	struct ospf6_neighbor *on;
 	struct listnode *node;
 
 	for (ALL_LIST_ELEMENTS_RO(oi->neighbor_list, node, on)) {
-		if (command == ZEBRA_BFD_DEST_REGISTER)
-			ospf6_bfd_info_nbr_create(oi, on);
-
+		/*
+		 * If not connected yet, don't create any session but defer it
+		 * for later.
+		 */
 		if (on->state < OSPF6_NEIGHBOR_TWOWAY) {
-			if (command == ZEBRA_BFD_DEST_DEREGISTER)
-				bfd_info_free(
-					(struct bfd_info **)&on->bfd_info);
+			bfd_sess_enable(on->bfd_session, false);
 			continue;
 		}
```

Comment:
Why do we need to disable the session? We just don't need to call `bfd_session_install` if the OSPF session is not established yet.

Reply:
I think we need to check for `on->bfd_session` being non-NULL here.

Reply:
Fixed: session always get the `bfd_session` allocated even when not state `full`.